opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -275,6 +289,29 @@ def set_status(self, status: trace_api.Status) -> None:
             return
         self.status = status
 
+    def __exit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        exc_val: Optional[BaseException],
+        exc_tb: Optional[TracebackType],
+    ) -> None:
+        """Ends context manager and calls `end` on the `Span`."""
+
+        if (
+            exc_val is not None
+            and self.status is None
+            and self._auto_update_status
+        ):
+            if isinstance(exc_val, StatusError):
+                self.set_status(
+                    Status(exc_val.canonical_code, exc_val.description)
+                )
+
+            else:
+                self.set_status(Status(StatusCanonicalCode.UNKNOWN))
```

Comment:
If `auto_update_status` is true, should the status be set to OK or UNKNOWN if there's no exception?

Reply:
Makes sense, will update.